packages/firestore/rollup.config.js

```diff
@@ -29,6 +29,22 @@ import { generateBuildTargetReplaceConfig } from '../../scripts/build/rollup_rep
 
 const util = require('./rollup.shared');
 
+// Customize how import.meta.url is polyfilled in cjs nodejs build. We use it to be able to use require() in esm.
+// It only generates the nodejs version of the polyfill, as opposed to the default polyfill which
+// supports both browser and nodejs. The browser support is unnecessary and doesn't work well with Jest. See https://github.com/firebase/firebase-js-sdk/issues/5687
+function importMetaUrlPolyfillPlugin() {
+  return {
+    name: 'import-meta-url-current-module',
+    resolveImportMeta(property, { moduleId }) {
+      if (property === 'url') {
+        // copied from rollup output
+        return `new (require('u' + 'rl').URL)('file:' + __filename).href`;
```

Comment:
what does separating the string do?

Reply:
It's what rollup outputs. I don't think it matters though. Updated